src/graphnet/pisa/fitting.py

```diff
@@ -74,6 +75,174 @@ def config_updater(
                 updater.write(configfile)
 
 
+class WeightFitter:
+    def __init__(
```

Comment:
A docstring here, or in `fit_weights`, would be helpful to explain which weights are being fitted, exactly.